sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Latest.java

```diff
@@ -153,11 +153,12 @@ public T extractOutput(TimestampedValue<T> accumulator) {
    * @param <K> The key type of the elements being combined.
    * @param <V> The value type of the elements being combined.
    */
-  public static <K, V> PTransform<PCollection<KV<K, V>>, PCollection<KV<K, V>>> perKey() {
+  public static <K, V> PerKey<K, V> perKey() {
     return new PerKey<>();
   }
 
-  private static class Globally<T> extends PTransform<PCollection<T>, PCollection<T>> {
+  /** Implementation of {@link #globally()}. */
+  public static class Globally<T> extends PTransform<PCollection<T>, PCollection<T>> {
```

Comment:
I may have missed the relevant section in the PTransform style guide -- why is it good to expose the classes if the user need not further configure them?